src/AdamWathan/Form/Elements/Checkbox.php

```diff
@@ -76,11 +76,14 @@ protected function setChecked($checked = true)
 
     protected function checkBinding()
     {
-        $currentValue = $this->getAttribute('value');
+        $currentValue = (string) $this->getAttribute('value');
```

Comment:
Can you explain all the string casting stuff in more detail and what the issue was that made you need to do it? It would be nice if it's avoidable.